### ISSUE TITLE
`dotDivide` changes to denseMatrix when the inputs are sparseMatrix

### ISSUE THREAD

**Issue by dvd101x** [josdejong/mathjs#3219]
Hi, 

I found an odd behavior of sparse matrices in `dotDivide` on 13.0.0

```js
math.dotDivide([1, 2], [1, 1])
// [1, 2]
// OK

math.dotDivide(math.sparse([1,2]), 1)
// SparseMatrix [1, 2]
// OK

math.dotDivide(math.sparse([1, 2]), math.sparse([1, 1]))
// DenseMatrix [[1], [2]]
// it's odd that the shape changes and it returns a DenseMatrix

math.dotDivide(math.sparse([1, 2]), math.sparse([1]))
// TypeError: undefined is not an object (evaluating 's[t]')
```

The implementation of `matrixAlgorithmSuite` in `dotDivide` seems different than the one in `dotMultiply` maybe it has something to do with the issue.

I found this while working on #3196, found that for sparse matrices broadcasting isn't working very well and went into a rabbit hole to find this separate issue.

**Comment by josdejong** [comment]
Thanks for reporting! We should indeed align this and make sure the output is a SparseMatrix for sparse inputs. I think we should mark it as a breaking change, since the current behavior may be depended upon.

**Comment by nkumawat34** [comment]
can i work on this issue.

**Comment by nkumawat34** [comment]
I found that in dotMultiply method algorithm which is included doing spare matrix dotMultiply sparse matrix give sparseMatrix
Algorithm is MatAlgo09xS0Sf
![dotMultiply](https://github.com/user-attachments/assets/fbbc6df6-8b12-4938-8458-7aa77c56db10)

In dotDivide method ,algorithm which is included doing spare matrix dotDivide sparse matrix gives DenseMatrix
ALgorithm is MatAlgo06xS0S0
![dotDivide](https://github.com/user-attachments/assets/4ea9670c-550e-4592-be18-b5dc28ecf748)

**Comment by josdejong** [comment]
Thanks Neeraj. 

I see this issue was picked up recently via https://github.com/josdejong/mathjs/pull/3287 by @rana-aakash so let's await a response there.

**Comment by rana-aakash** [comment]
Hello @josdejong 
Thanks for the reminder , I have submitted PR related to this at https://github.com/josdejong/mathjs/pull/3307

**Comment by josdejong** [comment]
Fixed in `v14.0.0`, thanks again.
